src/coreclr/jit/CMakeLists.txt

```diff
@@ -72,7 +72,7 @@ function(create_standalone_jit)
   endif ()
 endfunction()
 
-if (CLR_CMAKE_TARGET_ARCH_AMD64 OR CLR_CMAKE_TARGET_ARCH_ARM64 OR (CLR_CMAKE_TARGET_ARCH_I386 AND NOT CLR_CMAKE_HOST_UNIX))
+if (CLR_CMAKE_TARGET_ARCH_AMD64 OR CLR_CMAKE_TARGET_ARCH_ARM64 OR CLR_CMAKE_TARGET_ARCH_I386)
```

Comment:
Looks like this change was missed in #111861

Since floating->integral casts rely on at least baseline intrinsics support, `FEATURE_HW_INTRINSICS` is no longer optional on x86. Seems it's been building because the implementation was #ifdef`ed, but it would have failed at run time.